tests/framework/microvm.py

```diff
@@ -280,6 +280,24 @@ def setup(self):
         os.makedirs(self._kernel_path, exist_ok=True)
         os.makedirs(self._fsfiles_path, exist_ok=True)
 
+    @classmethod
+    def pin_vm_vcpus(cls):
+        """Pin the VM's threads on the last CPUs of the host."""
+        cpu_count = os.cpu_count()
+
+        # Returns a result with the following header:
+        # PID, SPID, TTY, TIME, CMD
+        _p = utils.run_cmd('ps -T -p $(pidof firecracker) | grep fc_vcpu')
+        assert _p.stderr == ''
+
+        vcpus = _p.stdout.split('\n')
+        vcpus = vcpus[0:-1]
+
+        for index, vcpu in enumerate(vcpus):
+            columns = vcpu.split()
+            pin_cpu = cpu_count - index - 1
+            _p = utils.run_cmd('taskset -cp {} {}'.format(pin_cpu, columns[1]))
+
```

Comment:
I have a couple of comments and questions here:

1. I think this method should be able to receive some parameters. Either the numa node on which it should pin the vcpus or the exact physical cpus. The problem is that jailer also does numa node pinning. And it pins the firecracker process to node 0 by default. If you always pin the vcpus to the last physical cpus (node > 0), this will create issues.
2. Can we do cpu shielding with `cset` instead of pinning ? It should lead to more predictible results.
3. Just to make sure we don't forget. We'll also have to setup NTP or PTP sync in the guest for more stable results. But we can do it in a future PR.

Reply:
1. Since we already have jailer which is capable of pinning the numa node I think it would be enought to have only pinning on the guest vcpus using the parameters and checking with the numa node used by jailer to make sure we avoid the actual error. What do you think?
2. Sure, this would much better.
3. I will add a TODO in the code too. Anyway, this would be very helpful in every integration test, I will try to do it as fast as I can.